### PR TITLE
[addons] cleanup C++ API headers

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

On the first commit so that it can be used for subsequent changes to other requests, the others to clean it up a little and also to use the new definitions.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/kodi-dev-kit/include/kodi/versions.h [hunk 1]
```diff
@@ -37,7 +37,7 @@
 // because cmake uses this area in this form to perform its addon dependency
 // check.
 // clang-format off
-#define ADDON_GLOBAL_VERSION_MAIN                     "1.3.0"
+#define ADDON_GLOBAL_VERSION_MAIN                     "1.3.1"
```
Comment (ksooo) [inline xbmc/addons/kodi-dev-kit/include/kodi/versions.h:40]: Why is this bump needed?

Comment (AlwinEsch) [reply xbmc/addons/kodi-dev-kit/include/kodi/versions.h:40]: Increase is related to the API changes but as `ADDON_GLOBAL_VERSION_MAIN_MIN` not changed, does all addons still works with it.

Comment (garbear) [reply xbmc/addons/kodi-dev-kit/include/kodi/versions.h:40]: Semantically, you can bump the patch version when you add a period, but the min version is where you start worrying about compatibility

**Review by ksooo — APPROVED** [review]
lgtm, your button, Alwin

**Comment by AlwinEsch** [comment]
jenkins build this with addons please

**Pull request merged**